stl/inc/utility

```diff
@@ -644,6 +645,77 @@ template <size_t _Idx>
 inline constexpr in_place_index_t<_Idx> in_place_index{};
 #endif // _HAS_CXX17
 
+#if _HAS_CXX20
+
+template <typename _Ty>
+inline constexpr bool _Is_integral_not_bool_char = is_integral_v<_Ty>
+    && !is_same_v<_Ty, bool> && !is_same_v<_Ty, char>
+    && !is_same_v<_Ty, char8_t> && !is_same_v<_Ty, char16_t>
+    && !is_same_v<_Ty, char32_t> && !is_same_v<_Ty, wchar_t>;
+
+// SAFE INTEGRAL COMPARISONS
+template <typename _Ty1, typename _Ty2>
+constexpr bool in_range(const _Ty2 _Value) noexcept {
+    static_assert(_Is_integral_not_bool_char<_Ty1> && _Is_integral_not_bool_char<_Ty2>);

```

Comment:
That would be superfluous. Unfortunately the paper is not yet there so I can only guess but my guess would be that this is a `Mandates:` not an `Expects:`. The former requires that the function is not part of the overload set.

Do you have the paper to confirm what the standard requires?

Reply:
The paper is not public available, but it's already merged in the current working draft repo. It is a mandates clause, as you can find [here](https://github.com/cplusplus/draft/blob/d34fb9a542c3a9873e49e53a8a86eccfdd336109/source/utilities.tex#L546-L548).

Reply:
I based on [draft](https://eel.is/c++draft/utility.intcmp) and [p0586r1](http://open-std.org/JTC1/SC22/WG21/docs/papers/2018/p0586r1.html)

```
For each of the following functions, if either of `T` or `U` is not
  a standard integer type or an extended integer type, as specified in 6.9.1,
  the call is ill-formed.
  [Note: std::byte, char, char16_t, char32_t, wchar_t, and bool are not
         comparable with these functions. --end note]
```

Reply:
P0586R2 now available http://open-std.org/JTC1/SC22/WG21/docs/papers/2020/p0586r2.html